src/slurm_plugin/clustermgtd.py

```diff
@@ -950,6 +963,21 @@ def _find_active_nodes(partitions_name_map):
                 active_nodes += partition.slurm_nodes
         return active_nodes
 
+    @staticmethod
+    def _get_unhealthy_ice_nodes(unhealthy_dynamic_nodes: list) -> dict:
+        """Get insufficient capacity compute resource and nodes, error code mapping."""
+        ice_compute_resources_and_nodes_map = {}
+        for node in unhealthy_dynamic_nodes:
+            if node.is_ice():
```

Comment:
would it make sense to filter out all static nodes here so that we don't iterate over them anymore?

Reply:
These are already dynamic nodes.
Do you mean in `_retrieve_scheduler_partitions`, we check if node is dynamic nodes before append it to compute_resource_nodes_map ?

Reply:
you are right these are already dynamic nodes sorry :) (unhealthy_DYNAMIC_nodes 😅)